sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -49,7 +49,7 @@ case class CachedRDDBuilder(
     storageLevel: StorageLevel,
     @transient cachedPlan: SparkPlan,
     tableName: Option[String])(
-    @transient private var _cachedColumnBuffers: RDD[CachedBatch] = null) {
+    @transient @volatile private var _cachedColumnBuffers: RDD[CachedBatch] = null) {
```

Comment:
Can this just be a lazy val? I don't see any caller that specifies this argument.

Reply:
IIUC we use `var` here to support cache buffer clearance? `def clearCache(blocking: Boolean = true)`

Reply:
@bersprockets @cloud-fan Oops, I just noticed this causes the Scala 2.11 build to fail:

```
[error] /home/jenkins/workspace/spark-master-test-maven-hadoop-2.7-ubuntu-scala-2.11/sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala:52: values cannot be volatile
[error]     @transient @volatile private var _cachedColumnBuffers: RDD[CachedBatch] = null) {
[error]   
```

It looks like this might be a scalac bug, that is only fixed in 2.12; didn't look too hard but ended up here:
https://github.com/scala/bug/issues/8873
https://github.com/scala/scala/pull/5294

It might be sufficient to move this to a private field, as I don't think any caller actually sets this value? Let me try.

Reply:
Thanks @srowen . Going forward, I assume I (and other contributors) should be building at least once with ```-Pscala-2.11``` before submitting or updating a PR?

Reply:
I wouldn't go that far as we have another job to check 2.11 after the fact and these are rare

Reply:
@bersprockets yeah that did it: https://github.com/apache/spark/pull/23864